### PR TITLE
[BEAM-8688] Upgrade GCSIO to 2.2.6

### PR DESCRIPTION
**Pull request by veblush**
Upgraded Google Cloud Storage client (GCSIO) to the latest 2.2.6 which contains recent bug fixes and improvements around gRPC implementation. This change adds the grpc-xds dependency because it started using the Traffic Director for Directpath and this should be transparent to users.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @suztomo `).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by veblush** [comment]
R: @suztomo 
R: @kennknowles

**Comment by veblush** [comment]
### Link check

```
$ ./gradlew -Ppublishing -PjavaLinkageArtifactIds=beam-sdks-java-io-google-cloud-platform,beam-runners-google-cloud-dataflow-java :checkJavaLinkage  
...
* What went wrong:
Execution failed for task ':sdks:java:io:hcatalog:compileJava'.
> Could not resolve all files for configuration ':sdks:java:io:hcatalog:compileClasspath'.
   > Could not find org.pentaho:pentaho-aggdesigner-algorithm:5.1.5-jhyde.
     Searched in the following locations:
       - file:/usr/local/google/home/veblush/git/beam/sdks/java/io/hcatalog/offline-repository/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - file:/usr/local/google/home/veblush/.m2/repository/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - https://public.nexus.pentaho.org/repository/proxy-public-3rd-party-release/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - https://oss.sonatype.org/content/repositories/staging/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - https://repository.apache.org/snapshots/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - https://repository.apache.org/content/repositories/releases/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
       - https://maven-central.storage-download.googleapis.com/maven2/org/pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde/pentaho-aggdesigner-algorithm-5.1.5-jhyde.pom
     Required by:
         project :sdks:java:io:hcatalog > org.apache.hive:hive-exec:2.1.0 > org.apache.calcite:calcite-core:1.6.0
```

Error doesn't seem to be relevant to this change.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -609,6 +609,8 @@ class BeamModulePlugin implements Plugin<Project> {
         grpc_netty                                  : "io.grpc:grpc-netty", // google_cloud_platform_libraries_bom sets version
         grpc_netty_shaded                           : "io.grpc:grpc-netty-shaded", // google_cloud_platform_libraries_bom sets version
         grpc_stub                                   : "io.grpc:grpc-stub", // google_cloud_platform_libraries_bom sets version
+        // Once grpc-xds is added to google_cloud_platform_libraries_bom, use google_cloud_platform_libraries_bom instead.
+        grpc_xds                                    : "io.grpc:grpc-xds:$grpc_version",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:613]: Declaring this key in the map doesn't have any effect.

**Comment by suztomo** [comment]
I haven't run the linkage check on Beam for a while.

Memo: org.pentaho:pentaho-aggdesigner-algorithm is [Spring Plugins](https://mvnrepository.com/repos/springio-plugins-release) repository (https://repo.spring.io/plugins-release/) as per https://mvnrepository.com/artifact/org.pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 1 above)
Comment (veblush) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:613]: Oh I added it to `sdks/java/io/google-cloud-platform/build.gradle`. PTAL

**Comment by veblush** [comment]
> I haven't run the linkage check on Beam for a while.
> 
> Memo: org.pentaho:pentaho-aggdesigner-algorithm is [Spring Plugins](https://mvnrepository.com/repos/springio-plugins-release) repository (https://repo.spring.io/plugins-release/) as per https://mvnrepository.com/artifact/org.pentaho/pentaho-aggdesigner-algorithm/5.1.5-jhyde.

I got the same error when running the linkage check on the master so I think it's not a regression caused by this PR.

**Comment by suztomo** [comment]
Run SQL_Java17 PreCommit

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run PostCommit_Java_DataflowV2

**Comment by veblush** [comment]
> @veblush The ticket ID in the PR suffix is the ticket that records dependency upgrades. Beam project automatically creates the tickets for each dependencies. In this case, it should be https://issues.apache.org/jira/browse/BEAM-8688 (This seems down now)

Tomo, the name of this PR starts with `[BEAM-8688]` and I thought it's enough. What else it needs to have? Should it be suffix? or a different format?

**Comment by veblush** [comment]
Oh, I thought I changed but it seems that you did. Thanks!

**Comment by suztomo** [comment]
My local invocation found dependency conflicts in gRPC classes.

```
Class io.grpc.ClientStreamTracer$InternalLimitedInfoFactory is not found;
Exception in thread "main"   referenced by 2 class files
    io.grpc.census.CensusStatsModule (io.grpc:grpc-census:1.43.2)
    io.grpc.census.CensusTracingModule (io.grpc:grpc-census:1.43.2)
  Cause:
    Dependency conflict: io.grpc:grpc-api:1.44.0 does not define Class io.grpc.ClientStreamTracer$InternalLimitedInfoFactory but io.grpc:grpc-api:1.43.2 defines it.
      selected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.39.0-SNAPSHOT / io.grpc:grpc-api:1.44.0 (compile
      unselected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.39.0-SNAPSHOT / org.apache.beam:beam-sdks-java-extensions-google-cloud-platform-core:2.39.0-SNAPSHOT (compile) / com.google.cloud.bigdataoss:gcsio:2.2.6 (compile) / io.grpc:grpc-census:1.43.2 (compile) / io.grpc:grpc-api:1.43.2 (compile)
    Dependency conflict: io.grpc:grpc-api:1.44.0 does not define Class io.grpc.ClientStreamTracer$InternalLimitedInfoFactory but io.grpc:grpc-api:1.43.2 defines it.
      selected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.39.0-SNAPSHOT / io.grpc:grpc-api:1.44.0 (compile)
      unselected: org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.39.0-SNAPSHOT / org.apache.beam:beam-sdks-java-extensions-google-cloud-platform-core:2.39.0-SNAPSHOT (compile) / com.google.cloud.bigdataoss:gcsio:2.2.6 (compile) / io.grpc:grpc-census:1.43.2 (compile) / io.grpc:grpc-api:1.43.2 (compile)
Problematic artifacts in the dependency tree:
io.grpc:grpc-census:1.43.2 is at:
  org.apache.beam:beam-sdks-java-io-google-cloud-platform:jar:2.39.0-SNAPSHOT / org.apache.beam:beam-sdks-java-extensions-google-cloud-platform-core:2.39.0-SNAPSHOT (compile) / com.google.cloud.bigdataoss:gcsio:2.2.6 (compile) / io.grpc:grpc-census:1.43.2 (compile)

```


https://gist.github.com/suztomo/7593d67629f61ce0395335bf911e6e43

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by veblush** [comment]
@suztomo Huh, that's interesting. Thanks for running the test. (I'm not sure why I couldn't catch it earlier with my test runs) https://github.com/grpc/grpc-java/pull/8768 looks relevant. and this is because grpc-census happens to have a different version of gRPC. Let's me add it as a new dependency as well.

**Comment by suztomo** [comment]
Nice. No new linkage errors https://gist.github.com/suztomo/0a2b2939876997dbc2312b69b9490104

**Pull request merged**